File: Week -3/Qs7.py
class Complex:
    def __init__(self, real=0.0, imag=0.0):
        self.real = real
        self.imag = imag

    def __add__(self, other):
        return Complex(self.real + other.real, self.imag + other.imag)

    def __sub__(self, other):
        return Complex(self.real - other.real, self.imag - other.imag)

    def __mul__(self, other):
        real_part = self.real * other.real - self.imag * other.imag
        imag_part = self.real * other.imag + self.imag * other.real
        return Complex(real_part, imag_part)

    def __str__(self):
        if self.imag >= 0:
            return f"{self.real:.2f} + {self.imag:.2f}i"
        else:
            return f"{self.real:.2f} - {-self.imag:.2f}i"

def read_complex():
    real = float(input("Enter the real part: "))
    imag = float(input("Enter the imaginary part: "))
    return Complex(real, imag)

def main():
    print("Enter the first complex number:")
    num1 = read_complex()
    print("Enter the second complex number:")
    num2 = read_complex()

    # Addition
    result = num1 + num2
    print(f"Sum: {result}")

    # Subtraction
    result = num1 - num2
    print(f"Difference: {result}")

    # Multiplication
    result = num1 * num2
    print(f"Product: {result}")

if __name__ == "__main__":
    main()

File: Week -3/Qs7.c
#include <stdio.h>

// Structure to represent a complex number
typedef struct {
    float real;
    float imag;
} Complex;

// Function to read a complex number
void readComplex(Complex *c) {
    printf("Enter the real part: ");
    scanf("%f", &c->real);
    printf("Enter the imaginary part: ");
    scanf("%f", &c->imag);
}

// Function to add two complex numbers
Complex addComplex(Complex c1, Complex c2) {
    Complex result;
    result.real = c1.real + c2.real;
    result.imag = c1.imag + c2.imag;
    return result;
}

// Function to subtract two complex numbers
Complex subtractComplex(Complex c1, Complex c2) {
    Complex result;
    result.real = c1.real - c2.real;
    result.imag = c1.imag - c2.imag;
    return result;
}

// Function to multiply two complex numbers
Complex multiplyComplex(Complex c1, Complex c2) {
    Complex result;
    result.real = c1.real * c2.real - c1.imag * c2.imag;
    result.imag = c1.real * c2.imag + c1.imag * c2.real;
    return result;
}

// Function to print a complex number
void printComplex(Complex c) {
    if (c.imag >= 0) {
        printf("%.2f + %.2fi\n", c.real, c.imag);
    } else {
        printf("%.2f - %.2fi\n", c.real, -c.imag);
    }
}

int main() {
    Complex num1, num2, result;

    // Read two complex numbers
    printf("Enter the first complex number:\n");
    readComplex(&num1);
    printf("Enter the second complex number:\n");
    readComplex(&num2);

    // Add the two complex numbers
    result = addComplex(num1, num2);
    printf("Sum: ");
    printComplex(result);

    // Subtract the two complex numbers
    result = subtractComplex(num1, num2);
    printf("Difference: ");
    printComplex(result);

    // Multiply the two complex numbers
    result = multiplyComplex(num1, num2);
    printf("Product: ");
    printComplex(result);

    return 0;
}
